vms/platformvm/state/state.go

```diff
@@ -2004,6 +2172,303 @@ func (s *state) writeExpiry() error {
 	return nil
 }
 
+// TODO: Update validator sets
+// TODO: Add caching
+func (s *state) writeSubnetOnlyValidators(updateValidators bool, height uint64) error {
```

Comment:
Note: this only works if the nodeIDs written here are unrelated to legacy subnet validators. This is fairly reasonable because we're going to be introducing 32-byte nodeIDs for SoVs.